bindings/go/plugin/manager/registries/componentversionrepository/converter.go

```diff
@@ -0,0 +1,230 @@
+		Version:    version,
+	}
+
+	return c.externalPlugin.GetComponentVersion(ctx, request, c.credentials)
+}
+
+func (c *componentVersionRepositoryWrapper) ListComponentVersions(ctx context.Context, component string) ([]string, error) {
+	request := ocmrepositoryv1.ListComponentVersionsRequest[runtime.Typed]{
+		Repository: c.repositorySpecification,
+		Name:       component,
+	}
+
+	return c.externalPlugin.ListComponentVersions(ctx, request, c.credentials)
+}
+
+func (c *componentVersionRepositoryWrapper) AddLocalResource(ctx context.Context, component, version string, res *descriptor.Resource, content blob.ReadOnlyBlob) (*descriptor.Resource, error) {
+	resources, err := descriptor.ConvertToV2Resources(c.scheme, []descriptor.Resource{*res})
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert resource: %w", err)
+	}
+
+	tmp, err := os.CreateTemp("", "resource")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp file: %w", err)
+	}
+	reader, err := content.ReadCloser()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read content: %w", err)
+	}
+
+	defer func() {
+		_ = tmp.Close()
+		_ = reader.Close()
+	}()
+
+	if _, err := io.Copy(tmp, reader); err != nil {
+		return nil, fmt.Errorf("failed to copy content: %w", err)
+	}
+
+	request := ocmrepositoryv1.PostLocalResourceRequest[runtime.Typed]{
+		Repository: c.repositorySpecification,
+		Name:       component,
+		Version:    version,
+		Resource:   &resources[0],
+		ResourceLocation: types.Location{
+			LocationType: types.LocationTypeLocalFile,
+			Value:        tmp.Name(),
+		},
+	}
+
+	return c.externalPlugin.AddLocalResource(ctx, request, c.credentials)
+}
+
+func (c *componentVersionRepositoryWrapper) GetLocalResource(ctx context.Context, component, version string, identity runtime.Identity) (blob.ReadOnlyBlob, *descriptor.Resource, error) {
+	request := ocmrepositoryv1.GetLocalResourceRequest[runtime.Typed]{
+		Repository: c.repositorySpecification,
+		Name:       component,
+		Version:    version,
+		Identity:   identity,
+	}
+
+	response, err := c.externalPlugin.GetLocalResource(ctx, request, c.credentials)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	rBlob, err := c.createBlobData(response.Location)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create blob data: %w", err)
+	}
+
+	convert := descriptor.ConvertFromV2Resources([]descriptorv2.Resource{*response.Resource})
+
+	return rBlob, &convert[0], nil
+}
+
+func (c *componentVersionRepositoryWrapper) AddLocalSource(ctx context.Context, component, version string, res *descriptor.Source, content blob.ReadOnlyBlob) (*descriptor.Source, error) {
+	sources, err := descriptor.ConvertToV2Sources(c.scheme, []descriptor.Source{*res})
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert source: %w", err)
+	}
+
+	tmp, err := os.CreateTemp("", "source")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp file: %w", err)
+	}
+	reader, err := content.ReadCloser()
```

Comment:
there should be an io utility function that will help you copy a blob to a file : `CopyBlobToOSPath`